scheduler/src/cook/mesos/api.clj

```diff
@@ -1884,6 +1885,25 @@
                                (str/join \space))}])
         true)))
 
+(defn check-retry-conflict
+  "Checks whether a 409 conflict should be returned during a retry operation. This should occur when one of the jobs:
+   - Is already completed
+   - Has already retried 'retries' times"
+  [conn ctx]
+  (let [jobs (::jobs ctx)
+        retries (get-in ctx [:request :body-params :retries])
+        db (d/db conn)]
+    (if (not (nil? retries))
+      (let [jobs-not-updated (filter (fn [uuid]
+                                       (let [job (d/entity (d/db conn) [:job/uuid uuid])]
+                                         (and (= :job.state/completed (:job/state job))
```

Comment:
Destructure `:job/state`?

Reply:
Discussed offline, I did the restructuring